src/_P001_Switch.ino

```diff
@@ -459,237 +446,263 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
+                PCONFIG(7) = 0;
+                PCONFIG_LONG(1) = millis();
+              }
 
-            currentStatus.state = state;
-            const boolean currentOutputState = currentStatus.output;
-            boolean new_outputState          = currentOutputState;
+              // just to simplify the reading of the code
+    #define COUNTER PCONFIG(7)
+    #define DC PCONFIG(4)
 
-            switch (PCONFIG(2))
-            {
-              case PLUGIN_001_BUTTON_TYPE_NORMAL_SWITCH:
-                new_outputState = state;
-                break;
-              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_LOW:
+              // check settings for doubleclick according to the settings
+              if ((COUNTER != 0) || ((COUNTER == 0) && ((DC == 3) || ((DC == 1) && (state == 0)) || ((DC == 2) && (state == 1))))) {
+                PCONFIG(7)++;
+              }
+    #undef DC
+    #undef COUNTER
 
-                if (!state) {
-                  new_outputState = !currentOutputState;
-                }
-                break;
-              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_HIGH:
+              currentStatus.state = state;
+              const boolean currentOutputState = currentStatus.output;
+              boolean new_outputState          = currentOutputState;
 
-                if (state) {
-                  new_outputState = !currentOutputState;
-                }
-                break;
-            }
+              switch (PCONFIG(2))
+              {
+                case PLUGIN_001_BUTTON_TYPE_NORMAL_SWITCH:
+                  new_outputState = state;
+                  break;
+                case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_LOW:
 
-            // send if output needs to be changed
-            if (currentOutputState != new_outputState)
-            {
-              byte output_value;
-              currentStatus.output = new_outputState;
-              boolean sendState = new_outputState;
+                  if (!state) {
+                    new_outputState = !currentOutputState;
+                  }
+                  break;
+                case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_HIGH:
 
-              if (Settings.TaskDevicePin1Inversed[event->TaskIndex]) {
-                sendState = !sendState;
+                  if (state) {
+                    new_outputState = !currentOutputState;
+                  }
+                  break;
               }
 
-              if ((PCONFIG(7) == 3) && (PCONFIG(4) > 0))
+              // send if output needs to be changed
+              if (currentOutputState != new_outputState)
               {
-                output_value = 3;                 // double click
-              } else {
-                output_value = sendState ? 1 : 0; // single click
-              }
-              event->sensorType = SENSOR_TYPE_SWITCH;
+                byte output_value;
+                currentStatus.output = new_outputState;
+                boolean sendState = new_outputState;
```

Comment:
type `boolean` is not a bool, so hard to tell what the negate operator will do here.
Can this be a bool?

Reply:
this is inside PLUGIN_WRITE. I haven't started looking at it yet.